### ISSUE TITLE
Can't use asStream with body on v8 but docs say we can

### ISSUE THREAD

**Issue by tomimarkus991** [elastic/elasticsearch-js#2241]
### Elasticsearch Version

v8.13

### Installed Plugins

@elastic/elasticsearch

### Java Version

_bundled_

### OS Version

Darwin ip-192-168-0-21.eu-north-1.compute.internal 23.0.0 Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000 arm64

### Problem Description

https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/8.13/as_stream_examples.html

Docs say to use body but there is no body on result.

using regular payload is not working

### Steps to Reproduce

try to add asStream to search and use body.

**Comment by JoshMock** [comment]
Thanks for the question @tomimarkus991. I think the issue here is that the code example was a bit confusing, as `result` is the value that should be operated on, not `body`. The return value of an API request is itself a `ReadableStream` when `asStream` is enabled. Unless you are also passing `meta: true`, in which case the response will be an object whose `body` property is a `ReadableStream`.

I updated the code example in https://github.com/elastic/elasticsearch-js/pull/2242 to make that more clear. I was able to run the updated code snippet verbatim against a local 8.13.0 Elasticsearch instance and it worked perfectly.
